### PR TITLE
Create add poem feature

### PR DESCRIPTION
**Pull request by Dalmano**
Relates #33

### CODE REVIEW

File: client/src/Component/Pages/AddPoem/index.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import Header from "../../CommonComponent/Header";
+import Input from "../../CommonComponent/Input";
+import Footer from "../../CommonComponent/Footer";
+
+const AddPoem = props => {
+  const [title, setTitle] = React.useState("");
+  const [poem, setPoem] = React.useState("");
+  const [error, setError] = React.useState(false);
+  const [typeError, setTypeError] = React.useState(false);
```
Comment (engshorouq) [inline client/src/Component/Pages/AddPoem/index.js]: I think there is no type in poem

File: client/src/Component/Pages/AddPoem/index.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  const [fetchError, setFetchError] = React.useState(false);
+  const [buttonContent, setButtonContent] = React.useState("Save");
+  const acceptedTypes = "";
```
Comment (engshorouq) [inline client/src/Component/Pages/AddPoem/index.js]: type for what? :thinking:

File: client/src/Component/Pages/AddPoem/index.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+  const message = {
+    default: "Please add a title and add a poem",
+    tyrebased:
+      "Please upload a different poem, we do not support that poem type",
+    fetch: "Sorry something went wrong, Please try again"
+  };
+
+  let type = poem.name.split(".")[poem.name.split(".").length - 1];
```
Comment (engshorouq) [inline client/src/Component/Pages/AddPoem/index.js]: I think for the poem just has title and content

File: client/src/Component/Pages/AddPoem/index.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      setError(true);
+    } else if (acceptedTypes.indexOf(type) === -1) {
+      setTypeError(true);
+    } else {
+      setError(false);
+      setTypeError(false);
+
+      setButtonContent("Loading...");
+
+      const poemData = new poemData();
```
Comment (engshorouq) [inline client/src/Component/Pages/AddPoem/index.js]: there is no class name is `poemData` Do you mean `FormData` ?? :thinking:

File: client/src/Component/Pages/AddPoem/index.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+              label="Title"
+              type="text"
+              action={e => {
+                setTitle(e.target.value);
+              }}
+            />
+            <label htmlFor="new-poem__text">Add a poem</label>
+            <input
+              id="mew-poem__text"
+              type="text"
```
Comment (engshorouq) [inline client/src/Component/Pages/AddPoem/index.js]: I think best if we put the type `textarea`

File: client/src/Component/Pages/AddPoem/index.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+              onChange={e => setPoem(e.target.poems[0])}
```
Comment (engshorouq) [inline client/src/Component/Pages/AddPoem/index.js]: mmmm there is `poems` inside the `target` :roll_eyes:

File: client/src/Component/Pages/AddPoem/index.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+            {error && <p className="add-poem__error">{message.default}</p>}
+            {typeError && (
+              <p className="add-poem__error">{message.tyrebased}</p>
+            )}
+            {fetchError && <p className="add-poem__error">{message.fetch}</p>}
+            <div className="add-poem__form__button">
+              <a href="/poems" className="large-skip__button">
+                Back
+              </a>
+              <button className="large-save__button" type="submit">
```
Comment (engshorouq) [inline client/src/Component/Pages/AddPoem/index.js]: I think there is `Button` Component why don't use it

File: client/src/Component/Pages/AddPoem/index.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+              type="text"
+              onChange={e => setPoem(e.target.poems[0])}
+            />
+            {error && <p className="add-poem__error">{message.default}</p>}
+            {typeError && (
+              <p className="add-poem__error">{message.tyrebased}</p>
+            )}
+            {fetchError && <p className="add-poem__error">{message.fetch}</p>}
+            <div className="add-poem__form__button">
+              <a href="/poems" className="large-skip__button">
```
Comment (engshorouq) [inline client/src/Component/Pages/AddPoem/index.js]: why use `a` tag also there is `Button` component and use `props.history.goBack()`

File: client/src/Component/Pages/AddPoem/index.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+  const [error, setError] = React.useState(false);
+  const [typeError, setTypeError] = React.useState(false);
+  const [fetchError, setFetchError] = React.useState(false);
+  const [buttonContent, setButtonContent] = React.useState("Save");
+  const acceptedTypes = "";
+
+  const message = {
+    default: "Please add a title and add a poem",
+    tyrebased:
+      "Please upload a different poem, we do not support that poem type",
```
Comment (engshorouq) [inline client/src/Component/Pages/AddPoem/index.js]: I think there is no uploading in this file. :thinking:

File: client/src/Component/Pages/AddPoem/index.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import Header from "../../CommonComponent/Header";
+import Input from "../../CommonComponent/Input";
+import Footer from "../../CommonComponent/Footer";
+
+const AddPoem = props => {
+  const [title, setTitle] = React.useState("");
+  const [poem, setPoem] = React.useState("");
+  const [error, setError] = React.useState(false);
+  const [typeError, setTypeError] = React.useState(false);
```
Comment (IsraaSulaiman) [inline client/src/Component/Pages/AddPoem/index.js]: This line is not necessary 
```suggestion
```

File: client/src/Component/Pages/AddPoem/index.js [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+  const message = {
+    default: "Please add a title and add a poem",
+    fetch: "Sorry something went wrong, Please try again"
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+
+    if ((title === "") | (poem === "")) {
+      console.log(title, poem);
```
Comment (IsraaSulaiman) [inline client/src/Component/Pages/AddPoem/index.js]:
```suggestion
```

File: client/src/Component/Pages/AddPoem/index.js [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+      setError(true);
+    } else {
+      setError(false);
+
+      setButtonContent("Loading...");
+
+      fetch("/api/v1/send-poem", {
```
Comment (IsraaSulaiman) [inline client/src/Component/Pages/AddPoem/index.js]: can you change the api url to `/api/v1/poems`? It is a best practise to use restful apis. 
Here is a link if you're interested:
https://restfulapi.net/resource-naming/

File: client/src/Component/Pages/AddPoem/index.js [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+              id="new-poem__title"
+              placeholder="Type here..."
+              label="Add a Title"
+              type="text"
+              action={e => {
+                setTitle(e.target.value);
+              }}
+            />
+
+            <Input
```
Comment (IsraaSulaiman) [inline client/src/Component/Pages/AddPoem/index.js]: As shorouq said, it is better to have textarea instead of input?

**Pull request merged**